eips_code/eof1_validation.py

```diff
@@ -0,0 +1,47 @@
+from eip4750 import * 
+from eip5450 import validate_function, FunctionType
+
+
+def read_eof1_header(container: bytes) -> (dict[int, list[int]], int):
+    section_sizes = {S_TYPE: [], S_CODE: [], S_DATA: []}
+    pos = len(MAGIC) + 1
+    while True:
+        section_id = container[pos]
+        pos += 1
+        if section_id == S_TERMINATOR:
+            break
+
+        section_sizes[section_id].append((container[pos] << 8) | container[pos + 1])
+        pos += 2
+
+    return section_sizes, pos
+
+def read_eof1_types(container: bytes, header_size: int, type_section_size: int) -> list[FunctionType]:
+    pos = header_size
+    types = []
+    while pos < header_size + type_section_size:
+        types.append(FunctionType(container[pos], container[pos + 1]))
+        pos += 2
+
+    return types
+
+
+# EOF code validation, including container format validation, code section validation and function stack validation.
```

Comment:
Maybe need to make this a bit more apparent as being the entry point.

Reply:
I moved it to the top of the file, does it help?